src/commands/index.ts

```diff
@@ -0,0 +1,19 @@
+import { ICommand } from './command.interface';
+import { PingCommand } from './ping.command';
+
+let commands: any = {};
+
+function registerCommand(command: ICommand) {
```

Comment:
For me this seems like - either we iterate the aliases in index and add them there or we iterate them during the Message Event. Just from my point of view it seems like it'd be smart to "register" them *once* and then iterate once each Event over them, instead of refering to the Command and getting it's aliases each time an Event fires and the iterating over them.  
It just *seems* like more operations for me, but if you'd prefer it that way, I can do. Will wait for a reply 😄!

Reply:
It is slightly more operations, except that I'd join all types of triggers into one array, and then iterate them that way.

A little more looping, yes, but I think it allows us to add more conditionals. What I mean is, if we do it this way, we're _kind of_ (but not technically) losing some of the properties which we may decide to use at a later date.

It seems cleaner to just iterate through all of the actual command objects, and check their criteria. If you want to avoid the extra operations/loops (and this might be cleaner), you could change `ICommand` into an abstract class with a function `shouldActivate(command: string, message?, args?)` which is implemented in command abstract class, which does the most basic shouldActivate checking (i.e. if any of the trigger conditions are satisfied).

Maybe this is extra, but potentially, it may also be worth having a way to pass on the actual string which met the trigger condition (i.e. matched the regex/etc).